File: SistemaRural_PE/src/modelo/AtencionMedica.java
package modelo;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class AtencionMedica {
	
 // Atributos de clase AtencionMedica
 private String idAtencion;
 private LocalDateTime fechaAtencion;
 private String diagnostico;
 private String tratamiento;

 private List<DetalleReceta> detallesReceta = new ArrayList<>();

 
 
 //Getters and setters
 public String getIdAtencion() {
	return idAtencion;
}

 public void setIdAtencion(String idAtencion) {
	this.idAtencion = idAtencion;
 }

 public LocalDateTime getFechaAtencion() {
	return fechaAtencion;
 }

 public void setFechaAtencion(LocalDateTime fechaAtencion) {
	this.fechaAtencion = fechaAtencion;
 }

 public String getDiagnostico() {
	return diagnostico;
 }

 public void setDiagnostico(String diagnostico) {
	this.diagnostico = diagnostico;
 }

 public String getTratamiento() {
	return tratamiento;
 }

 public void setTratamiento(String tratamiento) {
	this.tratamiento = tratamiento;
 }

 public List<DetalleReceta> getDetallesReceta() {
	return detallesReceta;
 }

 public void setDetallesReceta(List<DetalleReceta> detallesReceta) {
	this.detallesReceta = detallesReceta;
 }

 public AtencionMedica(String id, LocalDateTime fecha, String diag, String trat) {
     this.idAtencion = id;
     this.fechaAtencion = fecha;
     this.diagnostico = diag;
     this.tratamiento = trat;
 }

 public void agregarDetalleReceta(DetalleReceta detalle) {
     if(detalle.validarStock()) {
         detallesReceta.add(detalle);
         detalle.getMedicamento().actualizarStock(-detalle.getCantidad());
     }
 }

 public String generarInforme() {
     
     String medStr = detallesReceta.stream()
         .map(d -> d.getMedicamento().getNombre() + " (Cant: " + d.getCantidad() + ")")
         .collect(Collectors.joining(", "));
     return "Fecha: " + fechaAtencion + " | Diagnóstico Oculto (Privacidad) | Medicación: " + medStr;
 }
}

File: Clases de Python en el Sistema Rural/Atencion_Medica.py

class AtencionMedica:
    def __init__(self, id_atencion, fecha, diag, trat):
        self.id_atencion = id_atencion
        self.fecha_atencion = fecha
        self.diagnostico = diag
        self.tratamiento = trat
        self.detalles_receta = []

    def agregar_detalle_receta(self, detalle):
        if detalle.validar_stock():
            self.detalles_receta.append(detalle)
            detalle.medicamento.actualizar_stock(-detalle.cantidad)

    def generar_informe(self):
        med_str = ", ".join(
            [f"{d.medicamento.nombre} (Cant: {d.cantidad})" for d in self.detalles_receta]
        )
        return f"Fecha: {self.fecha_atencion.strftime('%Y-%m-%d %H:%M')} | Diagnóstico Oculto (Privacidad) | Medicación: {med_str}"